### PR TITLE
Add Backlog Provider

### PR DESCRIPTION
**Pull request by takashi-uesaka**
Backlog is ticket driven web service in Japan.

### CODE REVIEW

File: Owin.Security.Providers/Backlog/BacklogAuthenticationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+        /// <summary>
+        ///     Gets or sets the type used to secure data handled by the middleware.
+        /// </summary>
+        public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
+
+        /// <summary>
+        ///     Initializes a new <see cref="BacklogAuthenticationOptions" />
+        /// </summary>
+        public BacklogAuthenticationOptions()
```
Comment (jerriep) [inline Owin.Security.Providers/Backlog/BacklogAuthenticationOptions.cs:127]: Please add default value for CallbackPath in constructor:

```
CallbackPath = new PathString("/signin-backlog");
```

Comment (takashi-uesaka) [inline Owin.Security.Providers/Backlog/BacklogAuthenticationOptions.cs:127]: I added default value. Please check it out.

Comment (jerriep) [inline Owin.Security.Providers/Backlog/BacklogAuthenticationOptions.cs:127]: Thank you :)

**Comment by jerriep** [comment]
@takashi-uesaka We are looking for help moving to ASP.NET 5. Please read http://www.jerriepelser.com/blog/calling-contributors-for-oauth-providers

**Pull request merged**